### PR TITLE
Compile coffee script

### PR DESCRIPTION
**Pull request by DanielHeath**
Saw that https://github.com/benschwarz/bonsai/pull/30 hadn't been seeing much love and decided to finish the job.

### CODE REVIEW

File: lib/bonsai/exporter.rb [hunk 1]
```diff
@@ -112,19 +112,23 @@ def copy_assets
...
+            Bonsai.log "CSS Syntax error\n\n#{exception.message}"
+          end
         end
-      rescue Sass::SyntaxError => exception
-        Bonsai.log "CSS Syntax error\n\n#{exception.message}"
       end
-      
-      def remove_less_sass
-        Dir["#{path}/**/*.{less,sass}"].each{|f| FileUtils.rm(f) }
+
```
Comment (benschwarz) [inline lib/bonsai/exporter.rb:129]: I noticed in #30 that the cleanup task never removed scss files either. 
Probably worth adding a test for that too

File: lib/bonsai/exporter.rb [hunk 2]
```diff
@@ -112,19 +112,23 @@ def copy_assets
...
-      def generate_css
-        Dir["#{Bonsai.root_dir}/public/**/*.{less,sass,scss}"].each do |cssfile|
-          css = Tilt.new(cssfile).render
-          path = "#{File.dirname(cssfile)}/#{File.basename(cssfile, ".*")}.css"
-          
-          File.open(path, "w") {|file| file.write(css) }
+
+      def generate_assets
+        Dir["#{Bonsai.root_dir}/public/**/*.{less,sass,scss,coffee}"].each do |file|
+          begin
```
Comment (benschwarz) [inline lib/bonsai/exporter.rb:118]: any idea why we're using `begin` here rather than a lone `rescue` statement for the method?

Comment (DanielHeath) [inline lib/bonsai/exporter.rb:118]: By  moving the begin inside the block, we still generate an error but avoid skipping the rest of the files.
Previously, the broken stylesheet in the test fixtures meant that not all assets got compiled (which ones depend on the order your filesystem returns globs in)

Comment (benschwarz) [inline lib/bonsai/exporter.rb:118]: Good points, but who really wants it to keep generating if you've got
compilation errors?

Comment (DanielHeath) [inline lib/bonsai/exporter.rb:118]: If your workflow is 'make several changes' -> 'look at browser'?
It's easier to tell which file failed to generate if only one file is missing from the page.

Comment (benschwarz) [inline lib/bonsai/exporter.rb:118]: So if an error occurs, the file most likely won't be written / updated, and you'll need to check back in your terminal session to see the issue?   

Cheers,  

Ben

Comment (DanielHeath) [inline lib/bonsai/exporter.rb:118]: Yeah - whereas previously, if an error occurred, an (unpredictable) selection of your files won't be written/updated.

Comment (benschwarz) [inline lib/bonsai/exporter.rb:118]: Sounds good to me. I'll give this a test when I get the opportunity. Thanks for your contributions!  

Cheers,  

Ben

**Comment by DanielHeath** [comment]
Was just going through my dependencies and spotted this fork.
This has been waiting on a merge for a fair while now. Anything I can do to help it get merged?

**Comment by benschwarz** [comment]
Lets go with it! 
Thanks for your contributions, sorry I forgot all about them :-)

**Pull request merged**